### DISCUSSION TITLE
Authorization error when trying to update Team info

### DISCUSSION THREAD

**Discussion by mukeshwani** [outline/outline#3321] [Self Hosting]
I'm receiving an Authorization error on my self-hosted Outline Wiki (using Docker). I am using Keycloak for SSO, and Minio for Bucket storage. Everything other functionality work, except when trying to change the logo or update the Wiki name, I receive an Authorization error. Looking at the Browser dev tools, the call is being made to wiki.domain.tld/api/team.update

setup:
Docker / Ubuntu 20.04
Portainer 2.11.1
Outline stack
 - postgres:latest
 - minio/minio:latest
 - redis:latest
 - jboss/keycloak:latest
 - outlinewiki/outline:latest
Nginx Proxy Manager
 - SSL
  

Any ideas on what to check for this?

**Comment by tommoor** [comment]
But other API endpoints work fine? Honestly very strange – you have an admin account?

**Reply by mukeshwani** [reply]
Hi Tom, thanks for your reply. I am able to create Collections, docs etc so I believe other API endpoints are working.

When I check my profile in Outline, it doesn't show me as having any role (the role is blank) and so I am guessing my user is not an admin. I've tried to search for documentation on how to assign a role to my user or create a new user with the Admin role but I am not able to find any documentation on that.

Any idea on how to assign a role? I did go into Keycloak and create a role called 'admin' and 'admins' and assigned it to my user but no luck.

**Reply by tommoor** [reply]
Is there an existing admin on the team? Only they can make another user an admin – or you can edit the user row in the database directly.

**Reply by mukeshwani** [reply]
Hi Tom, I was able to resolve the issue. I can't pin-point the exact reason however here is what I did

1. In my docker-compose file, I had the wrong port for Minio. I was using port 9080:9080 (I changed this from the default since Portainer also runs on 9000). So I changed it to 9080:9000 which is the correct way.
2. In Keycloak I added the roles 'admin' and 'admins' (I didn't know which one is the right one so I created both and assigned my user to it).

After the above, I deleted the entire stack and started fresh. Whatever it was, it seems to have worked because now my User in Outline shows to have the role 'Admin' and I'm able to make the changes without error.

Thanks for the replies and help!
